### ISSUE TITLE
[Research] OUI Compliance audit of the `inspector` plugin

### ISSUE THREAD

**Issue by BSFishy** [opensearch-project/OpenSearch-Dashboards#4084]
Tracking issue for OUI compliance audit of the `inspector` plugin.

Follow the steps in #3945 to complete the audit, then open another issue with findings. Once the work of performing the audit is done, this issue can be closed.

If this issue isn't already assigned to someone and you'd like to take on this work, please ping @BSFishy in the comments and I can assign it to you.

**Comment by nurSaadat** [comment]
[src/plugins/inspector/public/views/_index.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/inspector/public/views/_index.scss) imports styles from [src/plugins/inspector/public/views/data/_index.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/inspector/public/views/data/_index.scss) and 
[src/plugins/inspector/public/views/requests/_index.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/inspector/public/views/requests/_index.scss).

[src/plugins/inspector/public/views/data/_index.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/inspector/public/views/data/_index.scss) imports styles from [src/plugins/inspector/public/views/data/_data_table.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/inspector/public/views/data/_data_table.scss).

[src/plugins/inspector/public/index.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/inspector/public/index.scss) imports styles from [src/plugins/inspector/public/views/_index.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/inspector/public/views/_index.scss).

[src/plugins/inspector/public/views/requests/_index.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/inspector/public/views/requests/_index.scss) imports styles from [src/plugins/inspector/public/views/requests/_requests.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/inspector/public/views/requests/_requests.scss).

As a dependency graph, it looks like this:

```mermaid
  graph TD;
      public/index.scss-->public/views/_index.scss;
      public/views/_index.scss-->public/views/data/_index.scss;
      public/views/_index.scss-->public/views/requests/_index.scss;
      public/views/data/_index.scss-->public/views/data/_data_table.scss;
      public/views/requests/_index.scss-->public/views/requests/_requests.scss;
```

Therefore, the main files to look at are [_requests.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/inspector/public/views/requests/_requests.scss), [inspector_panel.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/inspector/public/ui/inspector_panel.scss) and [_data_table.scss](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/src/plugins/inspector/public/views/data/_data_table.scss).

**Comment by nurSaadat** [comment]
_data_table.scss has only one class `insDataTableFormat__filter`. It is used to change the visibility of a EuiButtonIcon by changing its opacity on hover or on focus.

I think, this property is still needed, as OUI doesn't currently allow this type of behaviour for a button icon. 

Example can be seen here 
![button-icon-on-hover](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/39532643/c9f0518b-3fd7-4b16-8a34-5628be6e012f)

**Comment by nurSaadat** [comment]
`data_table.tsx` relies mostly on EUI components. The only component which is not from '@elastic/eui' is `FormattedMessage`. Should we wrap this component?

**Comment by nurSaadat** [comment]
`_requests.scss` contains three custom classes, all of them use euiSize variables to define properties.

I think, `insRequestSelector__singleRequest` may be substituted by OUI component. In .tsx file it is used to style a div
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/20c928fccf9b2516b36613874b335e9291b69f81/src/plugins/inspector/public/views/requests/components/request_selector.tsx#L163-173

**Comment by nurSaadat** [comment]
### Conclusion:
* OUI components and variables mostly are used throughout the plugin
* No need to remove custom class for EuiButtonIcon
* Remove `insInspectorPanel__flyoutBody` custom class

**Comment by nurSaadat** [comment]
@BSFishy could you please assign me this issue?

**Comment by joshuarrrr** [comment]
@nurSaadat When you get a chance, can you just copy/paste your audit to a separate issue?
